src/server/game/World/World.h

```diff
@@ -384,6 +384,13 @@ class World: public IWorld
 
     void RemoveOldCorpses();
 
+    /**
+     * @brief Executed when a World Session is being finalized. Be it from a normal login or via queue popping.
+     * 
+     * @param session The World Session that we are finalizing.
+     */
+    inline void FinalizePlayerWorldSession(WorldSession* session);
```

Comment:
```suggestion
    void FinalizePlayerWorldSession(WorldSession* session);
```

Reply:
Why? isn't it an optimization?

Reply:
I'm not sure, but I think such functions should be written entirely in a `.h` file

Reply:
You are correct if it's to be `inline` I should have moved the implementation into the header file too. 

Unless something is a super small one liner like a getter / setter I rarely like to put the implementation in .h files as a standard so i'd lean more towards simply not making this inline. It's an extremely minor compile time optimisation anyway.

However i'm happy to defer to whatever you to agree on considering this is my first contribution.

Reply:
It won't change that much though, since some compilers are able to decide which functions to inline when needed (when that keyword is not specified).

In this case I suggested to inline since this function is a private one that is only used internally to avoid code duplication: in few word we're defining a function to avoid writing the same code twice but since it's not used outside its own class, using the inline makes more sense to me. 

The optimization is not regarding the compiler, but the execution of the code. A non inlined function means that the CPU has to access the memory to get the address of that function and then execute it. When inlined (expanded) instead, the code to execute is already there

Reply:
It's a good practice to put inline functions inside a header only when you want to reuse that function in other cpp files.

Quoting from C++ FAQ:

```Note: It's imperative that the function's definition (the part between the {...}) be placed in a header file, unless the function is used only in a single .cpp file. In particular, if you put the inline function's definition into a .cpp file and you call it from some other .cpp file, you'll get an "unresolved external" error from the linker.```

Reply:
@kaytotes @Winfidonarleyan so what do you think?

Reply:
Switched to unresolved since I just noticed this function is inside the "public" list, it should be a private function since is not used outside and, based on what I said before, imho also inline (unless you have a technical reason why not doing it, other than a code-style opinion)

Reply:
I'll make the appropriate tweaks.

Reply:
@Yehonal I have moved the FinalizePlayerWorldSession function to be private since it won't be used elsewhere. I have also marked it as inline however based on the above quote from C++ FAQ since it will only used in World.cpp i've left the implementation in there. 

I did try to move the implementation into the `.h` file however the `sScriptMgr` instance usage made it a bit of a pain. Pulling `ScriptMgr.h` into `World.h` causes compilation issues. I also attempted to forward declare `ScriptMgr` so that I could instead do `ScriptMgr::instance()->` but that meant I'd then have to go around and include `ScriptMgr.h` everywhere that includes `World.h` and that became an issue.